vendor/bat-native-ledger/src/bat/ledger/internal/bat_contribution.cc

```diff
@@ -1032,42 +1033,50 @@ void BatContribution::GetContributeWinners(
     const unsigned int ballots,
     const std::string& viewing_id,
     const braveledger_bat_helper::PublisherList& list) {
-  unsigned int total_votes = 0;
-  std::vector<unsigned int> votes;
-  braveledger_bat_helper::Winners res;
+  braveledger_bat_helper::Winners winners =
+      GetStatisticalVotingWinners(ballots, list);
+  VotePublishers(winners, viewing_id);
+}
 
-  for (auto &item : list) {
-    if (item.percent_ <= 0) {
+bool BatContribution::GetStatisticalVotingWinner(
+    double dart,
+    const braveledger_bat_helper::PublisherList& list,
+    braveledger_bat_helper::WINNERS_ST* winner) {
+  double upper = 0.0;
+  for (const auto& item : list) {
+    upper += item.weight_;
+    if (upper < dart)
       continue;
-    }
 
-    braveledger_bat_helper::WINNERS_ST winner;
-    winner.votes_ = static_cast<unsigned int>(std::lround(
-        static_cast<double>(item.percent_) *
-        static_cast<double>(ballots) / 100.0));
+    winner->votes_ = 1;
+    winner->publisher_data_.id_ = item.id_;
+    winner->publisher_data_.duration_ = item.duration_;
+    winner->publisher_data_.score_ = item.score_;
+    winner->publisher_data_.visits_ = item.visits_;
+    winner->publisher_data_.percent_ = item.percent_;
+    winner->publisher_data_.weight_ = item.weight_;
```

Comment:
Side note (not important for this change, since this was effectively in the original anyway): Can we just do `winner->publisher_data_ = item` instead here?

Reply:
Yes, good idea.